app/services/candidate_api/serializers/v1_2.rb

```diff
@@ -0,0 +1,128 @@
+          }
+        end
+      end
+
+      def query
+        Candidate
+        .left_outer_joins(:application_forms)
+        .where(application_forms: { recruitment_cycle_year: RecruitmentCycle.current_year })
+        .or(Candidate.where('candidates.created_at > ? ', CycleTimetable.apply_1_deadline(RecruitmentCycle.previous_year)))
+        .distinct
+        .includes(application_forms: { application_choices: %i[provider course interviews], application_references: [] })
+        .where('candidate_api_updated_at > ?', updated_since)
+        .order('candidates.candidate_api_updated_at DESC')
+      end
+
+    private
+
+      def serialize_references(application_form)
+        {
+          completed: application_form.references_completed,
+          data:
+            application_form.application_references.order(:id).map do |reference|
+              {
+                id: reference.id,
+                requested_at: reference.requested_at&.iso8601,
+                feedback_status: reference.feedback_status,
+                referee_type: reference.referee_type,
+              }
+            end,
+        }
+      end
+
+      def serialize_qualifications(application_form)
+        {
+          completed: application_form.qualifications_completed?,
+        }
+      end
+
+      def serialize_personal_statement(application_form)
+        {
+          completed: application_form.becoming_a_teacher_completed,
+        }
+      end
+
+      def serialize_application_choices(application_form)
+        {
+          completed: application_form.course_choices_completed,
+          data:
+            application_form.application_choices.order(:id).map do |application_choice|
+              {
+                id: application_choice.id,
+                created_at: application_choice.created_at&.iso8601,
+                updated_at: application_choice.updated_at&.iso8601,
+                status: application_choice.status,
+                provider: serialize_provider(application_choice.provider),
+                course: serialize_course(application_choice.course),
+                interviews: serialize_interviews(application_choice),
+              }
+            end,
+        }
+      end
+
+      def serialize_provider(provider)
+        {
+          name: provider.name,
+        }
+      end
+
+      def serialize_course(course)
+        {
+          uuid: course.uuid,
+          name: course.name,
+        }
+      end
+
+      def serialize_interviews(application_choice)
+        application_choice.interviews.map do |interview|
+          serialize_interview(interview)
+        end
+      end
+
+      def serialize_interview(interview)
+        {
+          id: interview.id,
+          date_and_time: interview.date_and_time&.iso8601,
+          created_at: interview.created_at&.iso8601,
+          updated_at: interview.updated_at&.iso8601,
+          cancelled_at: interview.cancelled_at&.iso8601,
+        }
+      end
+    end
+  end
+end
```

Comment:
Is it worth it for both serializers to share code via a module or inheritance? Wdyt?

Reply:
I've extracted a `Base` class. It's fairly small but worth having in place I think.